include/appbase/application_base.hpp

```diff
@@ -145,9 +144,31 @@ class application_base {
             }
          }
 
+         quit();
+
          try {
-            exec.clear(); // make sure the queue is empty
-            shutdown();   // may rethrow exceptions
+            shutdown_plugins();   // may rethrow exceptions
+         } catch (...) {
+            if (!eptr)
+               eptr = std::current_exception();
+         }
+
+         work.reset();
+
+         try {
+            // plugins shutdown down at this point,
+
+            // Drain the io_service of anything that could be referencing plugins.
+            // Note this does not call exec.execute_highest(), so only drains into the priority queue assuming nothing
+            // has hijacked the io_service for other purposes.
+            io_serv.restart();
+            while (io_serv.poll())
+               ;
+            // clear priority queue of anything pushed by poll()
+            exec.clear();
+
+            // destroy the plugins now that all lambda that reference them have been destroyed
```

Comment:
Does this really destroy lambdas that reference plugins?

Reply:
This destroys the plugins **now that all lambda that** ..